flink-connector-bigquery-examples/src/main/java/org/apache/flink/examples/gcp/bigquery/BigQueryExample.java

```diff
@@ -0,0 +1,140 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A simple BigQuery table read example with Flink's DataStream API.
+ *
+ * <p>The Flink pipeline will try to read the specified BigQuery table, potentially limiting the
+ * element count to the specified row restriction and limit count, returning {@link GenericRecord}
+ * representing the rows, to finally prints out some aggregated values given the provided payload's
+ * field.
+ *
+ * <p>Note on row restriction: In case of including a restriction with a temporal reference,
+ * something like {@code "TIMESTAMP_TRUNC(ingestion_timestamp, HOUR) = '2023-06-20 19:00:00'"}, and
+ * launching the job from Flink's Rest API is known the single quotes are not supported and will
+ * make the pipeline fail. As a workaround for that case using \u0027 as a replacement will make it
+ * work, example {@code "TIMESTAMP_TRUNC(ingestion_timestamp, HOUR) = \u00272023-06-20
+ * 19:00:00\u0027"}.
+ */
+public class BigQueryExample {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryExample.class);
+
+    public static void main(String[] args) throws Exception {
+        // parse input arguments
+        final ParameterTool parameterTool = ParameterTool.fromArgs(args);
+
+        if (parameterTool.getNumberOfParameters() < 4) {
+            LOG.error(
+                    "Missing parameters!\n"
+                            + "Usage: flink run <additional runtime params> BigQuery.jar"
+                            + " --gcp-project <gcp-project> --bq-dataset <dataset name>"
+                            + " --bq-table <table name> --agg-prop <payload's property>"
+                            + " --restriction <single-quoted string with row predicate>"
+                            + " --limit <optional: limit records returned>");
+            return;
+        }
+
+        String projectName = parameterTool.getRequired("gcp-project");
+        String datasetName = parameterTool.getRequired("bq-dataset");
+        String tableName = parameterTool.getRequired("bq-table");
+        String rowRestriction = parameterTool.get("restriction", "").replace("\\u0027", "'");
+        Integer recordLimit = parameterTool.getInt("limit", -1);
+        String recordPropertyToAggregate = parameterTool.getRequired("agg-prop");
+
+        runFlinkJob(
+                projectName,
+                datasetName,
+                tableName,
+                recordPropertyToAggregate,
+                rowRestriction,
+                recordLimit);
+    }
+
+    private static void runFlinkJob(
+            String projectName,
+            String datasetName,
+            String tableName,
+            String recordPropertyToAggregate,
+            String rowRestriction,
+            Integer limit)
+            throws Exception {
+
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.enableCheckpointing(60000L);
```

Comment:
Can we pass this as a user provided configuration and have this value as a default if user does not pass the value

Reply:
I meant `60000L` can be passed as a configuration instead of hardcoding. So that users can run this with different checkpoint intervals